core/config/project_settings.cpp

```diff
@@ -1276,12 +1276,12 @@ ProjectSettings::ProjectSettings() {
 	custom_prop_info["display/window/size/mode"] = PropertyInfo(Variant::INT, "display/window/size/mode", PROPERTY_HINT_ENUM, "Windowed,Minimized,Maximized,Fullscreen,Exclusive Fullscreen");
 
 	// Keep the enum values in sync with the `DisplayServer::SCREEN_` enum.
-	GLOBAL_DEF_BASIC("display/window/size/initial_screen", -2);
-	String screen_hints = "Primary Monitor:-2"; // Note: Main Window Monitor:-1 is not used for the main window, skip it.
-	for (int i = 0; i < 64; i++) {
-		screen_hints += ",Monitor " + itos(i + 1) + ":" + itos(i);
-	}
-	custom_prop_info["display/window/size/initial_screen"] = PropertyInfo(Variant::INT, "display/window/size/initial_screen", PROPERTY_HINT_ENUM, screen_hints);
+	GLOBAL_DEF_BASIC("display/window/size/initial_position_type", 1);
+	GLOBAL_DEF_BASIC("display/window/size/initial_position", Vector2i());
+	GLOBAL_DEF_BASIC("display/window/size/initial_screen", 0);
+	custom_prop_info["display/window/size/initial_position_type"] = PropertyInfo(Variant::INT, "display/window/size/initial_position_type", PROPERTY_HINT_ENUM, "Absolute,Primary Screen Center,Other Screen Center");
+	custom_prop_info["display/window/size/initial_position"] = PropertyInfo(Variant::VECTOR2I, "display/window/size/initial_position");
+	custom_prop_info["display/window/size/initial_screen"] = PropertyInfo(Variant::INT, "display/window/size/initial_screen", PROPERTY_HINT_RANGE, "0,64,1,or_greater");
```

Comment:
`GLOBAL_DEF*` now supports `PropertyInfo` directly insteaad of just the property name, so you could unify these.

(Same goes for other uses of `custom_prop_info` in this file, but that's outside the scope of this PR.)